spec/models/user_spec.rb

```diff
@@ -0,0 +1,78 @@
+require "rails_helper"
+
+RSpec.describe User, type: :model do
+
+  it "has a valid user" do
+    expect(build(:user)).to be_valid
+  end
+
+  describe "associations" do
+
+    it "has many bookings" do
+      is_expected.to have_many(:bookings).dependent :destroy
+    end
+
+    it "has many likes" do
+      is_expected.to have_many(:likes).dependent :destroy
+    end
+
+    it "has many reviews" do
+      is_expected.to have_many(:reviews).dependent :destroy
+    end
+
+    it "has many comments" do
+      is_expected.to have_many(:comments).dependent :destroy
+    end
+
+    context "presence" do
+      it { should validate_presence_of :name }
+      it { should validate_presence_of :phonenumber }
+    end
+  end
+
+  describe "roles" do
+    it "are user or admin" do
+      is_expected.to define_enum_for(:role).with_values %i(user admin)
+    end
+  end
+
+  describe "validate" do
+    let(:user){FactoryBot.create :user}
+
+    describe "name" do
+      it { is_expected.to validate_presence_of(:name) }
+      it "can't be blank" do
+        user.name = nil
+        user.valid?
+        expect(user.errors.messages[:name].first).to eq(
+          I18n.t "rspec.error_messages.blank")
+      end
+    end
+
+    describe "phonenumber" do
+      it { is_expected.to validate_presence_of(:phonenumber) }
+      it { should validate_numericality_of(:phonenumber) }
+
+      it "when phonenumber is too short" do
+        user.phonenumber = Faker::Number.number 8
+        user.valid?
+        expect(user.errors.messages[:phonenumber].first).to eq(
+          I18n.t "rspec.error_messages.min_length_phonenumber", size: 9)
+      end
+
+      it "when phonenumber is too long" do
+        user.phonenumber = Faker::Number.number 12
+        user.valid?
+        expect(user.errors.messages[:phonenumber].first).to eq(
+          I18n.t "rspec.error_messages.max_length_phonenumber", size: 11)
```

Comment:
1. giá trị 11, em get từ file setting hì

validate theo kiểu như bên dưới - em tìm hiểu thêm về I18n validate message
it {is_expected.to validate_length_of(:phonenumber).is_at_most(Settings.phone_max_length)
      .with_message(I18n.t("activerecord.errors.models.user.attributes.phone.too_long"))}

Reply:
Dạ em có thử theo cách này rồi chị. Vì phonenumber em có validates numericality: true nên nó không cho dùng kiểu này ạ.